app/src/org/commcare/fragments/CommCarePreferenceFragment.java

```diff
@@ -171,9 +171,25 @@ protected boolean isPersistentAppPreference() {
 
     @Override
     public void onSharedPreferenceChanged(SharedPreferences sharedPreferences, String key) {
-        String prefValue = sharedPreferences.getString(key, null);
+        String prefValue = getPrefValueAsString2(sharedPreferences, key);
         if (prefValue != null) {
             FirebaseAnalyticsUtil.reportEditPreferenceItem(key, prefValue);
         }
     }
+
+    private static String getPrefValueAsString(SharedPreferences sharedPreferences, String key) {
+        try {
+            return sharedPreferences.getString(key, null);
+        } catch (ClassCastException e) {
+            try {
+                return String.valueOf(sharedPreferences.getBoolean(key, false));
+            } catch (ClassCastException e2) {
+                return String.valueOf(sharedPreferences.getInt(key, -1));
+            }
+        }
+    }
+
+    private static String getPrefValueAsString2(SharedPreferences sharedPreferences, String key) {
+        return String.valueOf(sharedPreferences.getAll().get(key));
```

Comment:
@wpride @ctsims What about this instead? (I just tested that it works with both a String and non-String pref value)

Reply:
Looks great, nice call